packages/authorization/src/types.ts

```diff
@@ -187,4 +187,9 @@ export interface AuthorizationOptions {
    * not associated with authorization metadata.
    */
   defaultMetadata?: AuthorizationMetadata;
+  /**
+   * Default HTTP status code when the final decision is `AuthorizationDecision.DENY`.
+   * If not set, default to 403
+   */
+  defaultStatusCodeForDeny?: number;
```

Comment:
Do we want to make it even more stricter, such as `401 | 403`?

Reply:
I wouldn't be that strict. I can imagine a valid scenario for returning a `404`. For example, a system might have `/users/:id` returning `404` to prevent malicious agents from mining existing user ids.
Also some systems occasionally implement their own extensions in `4xx` space for whatever reasons they have.

I am happy with anything other than `401` as `401` usually triggers a token refresh and it is important to distinguish it from non-retriable failures to avoid loops.